students/jayjohnson/lesson06/mailroom_final.py

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python3
+'''
+Dwight Johnson - MailRoom - Final
+'''
+import sys
+from textwrap import dedent  # nifty utility!
+import math
+
+
+class MailRoom:
+    def __init__(self, name, balance=0.0):
+        self.name = name
+        self._balance = []
+        self._balance.append(float(balance))
+
+    def deposit(self, amount):
+        """make a deposit"""
+        self._balance.append(float(amount))
+
+    def donation_sum(self):
+        """sum donation totals together"""
+        return print("Total donations: $" + str(sum(self._balance)))
+
+    # def create_donor_list(self):
+    #     return
+
+    @property
+    def balance(self):
+        """check the donation balance"""
+        return self._balance
+
+    @property
+    def identity(self):
+        """check the name"""
+        return self.name
+
+    # offical string repesentation
+    def __repr__(self):
+        return '{0.__class__.__name__}(name={0.name}, balance={0.balance})'.format(self)
+
+    # informal string representation
+    def __str__(self):
+        return 'Donation totals for {}, current donations: {}'.format(self.name, self.balance)
+
+    def __name__(self):
+        return self.name
+
+#print(donor2)
+#print(donor2.balance)
+#donor2.deposit(1000)
+# print(str(donor1))
+# print(repr(donor1))
+# # print all donors
+# print(donor1)
+#print(donor2)
+# print(donor3)
+
+
+#
+# def report():
+#     print(donor1)
+#     print(donor2)
+#     print(donor3)
+#def print_donors():
+    #print(donor_new)
+#print_donors()
+
+
+def mainloop():
+
+    #build donorlist
+    donor1 = MailRoom('BILL', 100.00)
+    donor2 = MailRoom('Alex', 2000.00)
+    donor3 = MailRoom('Sam', 300000.00)
+    donor1.deposit(50.00)
+    donor1.deposit(20.00)
+    donor2.deposit(5000.00)
+    donor2.deposit(10000.00)
+    donor3.deposit(1000000.00)
+    donor3.deposit(50000.00)
+
+    donor_db = []
+
+    # add donors to database
+    donor_db.append(donor1)
+    donor_db.append(donor2)
+    donor_db.append(donor3)
```

Comment:
Ideally we wanted two classes - one for donor and another for all donors. This isn't quite the answer. I encourage you to check out what Hosung put in his directory to get an idea of what that would look like